mods/mc_tutorial/init.lua

```diff
@@ -0,0 +1,1708 @@
+----- update the record_fs menu so that on_completion items and tools are displayed in an inventory and the number of items given can be set by the player recording the tutorial
+----- add option to display a message after completing a specific action, like "now do this next"\
+
+mc_tutorial = {
+    path = minetest.get_modpath("mc_tutorial"),
+    tutorials = minetest.get_mod_storage(),
+    record = {
+        active = {},
+        temp = {},
+        edit = {},
+        listener = {
+            wield = {},
+            key = {}
+        },
+        timer = {}
+    },
+    fs_context = {},
+    active = {},
+
+    ACTION = { -- action constants
+        PUNCH = 1,
+        DIG = 2,
+        PLACE = 3,
+        WIELD = 4,
+        KEY = 5,
+        LOOK_YAW = 6,
+        LOOK_PITCH = 7,
+        LOOK_DIR = 8,
+        POS_ABS = 9,
+        GROUP = 10,
+        -- WIP beyond this point
+        POS_REL = 11, 
+        MSG_CHAT = 12,
+        MSG_POPUP = 13,
+        INV_PUT = 14,
+        INV_TAKE = 15,
+        INV_MOVE = 16,
+        CRAFT = 17,
+    },
+    GROUP = { -- group type constants
+        START = 1,
+        END = 2
+    },
+    SIDEBAR = {
+        NONE = 0,
+        ITEM = 1,
+        KEY = 2
+    },
+    DIR = {
+        VECTOR = 1,
+        YAW_PITCH = 2,
+    }
+}
+-- local constants
+local RW_LIST, RW_SELECT = 1, 2
+
+local function get_context(player)
+    local pname = (type(player) == "string" and player) or (player:is_player() and player:get_player_name()) or ""
+    mc_tutorial.fs_context[pname] = mc_tutorial.fs_context[pname] or {
+        clear = function(self)
+            mc_tutorial.fs_context[pname] = nil
+        end
+    }
+    return mc_tutorial.fs_context[pname]
+end
+
+-- Store and load default settings in the tutorial_settings.conf file
+local settings = Settings(mc_tutorial.path .. "/tutorial_settings.conf")
+function mc_tutorial.fetch_setting(name)
+    local sname = "mc_tutorial." .. name
+    return settings and settings:get(sname) or minetest.settings:get(sname)
+end
+function mc_tutorial.fetch_setting_table(name)
+    local setting_string = mc_tutorial.fetch_setting(name)
+    local table = mc_helpers.split(setting_string, ",")
+    for i,v in ipairs(table) do
+        table[i] = v:trim()
+    end
+    return table
+end
+function mc_tutorial.fetch_setting_key_table(name)
+    local output = {}
+    for _,priv in pairs(mc_tutorial.fetch_setting_table(name) or {}) do
+        output[priv] = true
+    end
+    return output
+end
+
+mc_tutorial.player_priv_table = mc_tutorial.fetch_setting_key_table("player_priv_table") or {interact = true}
+mc_tutorial.recorder_priv_table = mc_tutorial.fetch_setting_key_table("recorder_priv_table") or {teacher = true, interact = true}
+mc_tutorial.check_interval = math.max(tonumber(mc_tutorial.fetch_setting("check_interval")), 0.1) or 1
+mc_tutorial.check_dir_tolerance = mc_tutorial.fetch_setting("check_dir_tolerance") or 0.01745
```

Comment:
That might actually work better as a separate action (`LOOK_NODE`), since using a node for comparison would mean that the player could look at it from any direction to complete the action.

Reply:
I'll implementing this along with the other new tutorial actions as part of an upcoming PR. right now, it's more important to get known bugs fixed and in time for the upcoming user tests.